railties/lib/rails/commands/console/console_command.rb

```diff
@@ -12,6 +12,50 @@ def filter_backtrace(bt)
       end
     end
 
+    class IRBConsole
+      def initialize
+        require "irb"
+        require "irb/completion"
+        IRB::WorkSpace.prepend(BacktraceCleaner)
+
+        if !Rails.env.local?
+          # Use env var here so users can override them with env var too
+          ENV["IRB_USE_AUTOCOMPLETE"] ||= "false"
+        end
+      end
+
+      def name
+        "IRB"
+      end
+
+      def start
+        IRB.setup(nil)
+
+        env =
+          case Rails.env
+          when "development"
+            IRB::Color.colorize("dev", [:GREEN])
+          when "test"
+            IRB::Color.colorize("test", [:GREEN])
+          when "production"
+            IRB::Color.colorize("prod", [:RED])
```

Comment:
If colorising is not available, like `TERM=dumb` or IRB's coloring is disabled, this method simply returns the original string.

Also, it's pretty hard to test the colorised result here because `TERM=dumb` is required for Reline to function correctly in a PTY test environment. But in IRB we have quite [some tests for it](https://github.com/ruby/irb/blob/master/test/irb/test_color.rb).

Reply:
Ok this is now covered in unit tests too.